src/main/java/com/rental/carshowroom/controller/LeasingController.java

```diff
@@ -1,8 +1,78 @@
 package com.rental.carshowroom.controller;
 
-import org.springframework.web.bind.annotation.RequestMapping;
+import com.rental.carshowroom.model.Leasing;
+import com.rental.carshowroom.model.enums.LeasingStatus;
+import com.rental.carshowroom.service.LeasingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
+import javax.validation.Valid;
+import java.math.BigDecimal;
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.List;
+@RestController
 @RequestMapping("/api/leasing")
 public class LeasingController {
 
+    private LeasingService leasingService;
+
+    @Autowired
+    public LeasingController(LeasingService leasingService) {
+        this.leasingService = leasingService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<Leasing>> showAll() {
+        return ResponseEntity.ok(leasingService.showAllLeasings());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Leasing> getLeasingById(@PathVariable Long id) {
+        return ResponseEntity.ok(leasingService.findLeasing(id));
+    }
+
+    @PostMapping
+    public ResponseEntity<Leasing> addLeasing(@RequestBody @Valid Leasing leasing) {
```

Comment:
return payment or null if initial payment is null -> leasing may be taken from URI